### PR TITLE
feat: cache iOS dependencies

### PR DESCRIPTION
**Pull request by victorrabelozup**
## Description

Changed the iOS PR check action in the CI to cache dependencies from SPM. The intension was to gain performance.
Also removed Carthage action because we don't use it as dependency manager anymore.

## Related Issues

#522 

## Tests

Tested by checking the time used to fetch dependencies and looking if the cache step could get data from cache folder.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [DCO].
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

[issue database]: https://github.com/ZupIT/beagle/issues
[DCO]: https://developercertificate.org/

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: .github/workflows/ios-pr.yml [hunk 1]
```diff
@@ -25,8 +25,14 @@ jobs:
...
-            -   name: install Carthage
-                run: brew install carthage
+            -   name: Cache SPM dependencies
+                uses: actions/cache@v2
+                env:
+                    spm-cache-key: ${{ runner.os }}-spm-${{ hashFiles('iOS/**/Package.resolved') }}
+                    spm-path: ~/**/DerivedData
+                with:
+                    path: ${{ env.spm-path }}
+                    key: ${{ env.spm-cache-key }}
```
Comment (theffc) [inline .github/workflows/ios-pr.yml:35]: Couldn't we use it like this?
```suggestion
                with:
                    path: ${{ runner.os }}-spm-${{ hashFiles('iOS/**/Package.resolved') }}
                    key: ~/**/DerivedData
```

Comment (victorrabelozup) [reply .github/workflows/ios-pr.yml]: I was following the backend example, also I thought that with env variables would be more organized. But we can do this way.

**Comment by theffc** [comment]
it seems like our ci cache is just working when SPM downloads dependencies source code, but not for actually caching build artifacts. We will take a closer look to that and how many data is actually using on the following days.

**Pull request merged**